pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+                raise MisconfigurationException(
+                    f"Precision {repr(precision)} is invalid. "
+                    f"Allowed precision values: {PrecisionType.supported_types()}"
+                )
+            self._precision_flag = precision
+            # handle duplications and conflict
+            # [RFC] current logic doesn't handle precision_plugin duplication
+            # if plugins:
+            #     for plugin in plugins:
+            #         if isinstance(plugin, PrecisionPlugin):
+            #             self._precision_flag = precision
+            # raise MisconfigurationException("precision set in both precision flag and plugin flag")
+
+        if plugins:
+            for plugin in plugins:
+                if isinstance(plugin, Strategy) or isinstance(plugin, str) and plugin in self._existing_strategies_str:
+                    self._strategy_flag = plugin
+                    rank_zero_deprecation(
+                        f"Passing {plugin} `strategy` to the `plugins` flag in Trainer has been deprecated"
+                        f" in v1.5 and will be removed in v1.7. Use `Trainer(strategy={plugin})` instead."
+                    )
 
-        for plug in self.plugins:
-            if isinstance(plug, str) and plug in StrategyRegistry:
-                if strategy is None:
-                    strategy = StrategyRegistry.get(plug)
+                elif isinstance(plugin, PrecisionPlugin) or (
+                    isinstance(plugin, str) and plugin in self._supported_precision
+                ):
+                    self._precision_flag = plugin
+                elif isinstance(plugin, CheckpointIO):
+                    self.checkpoint_io = plugin
+                elif isinstance(plugin, ClusterEnvironment):
+                    self._cluster_environment = plugin
                 else:
                     raise MisconfigurationException(
-                        "You can only specify one precision and one training type plugin."
-                        " Found more than 1 training type plugin:"
-                        f' {StrategyRegistry[plug]["strategy"]} registered to {plug}'
+                        f"Found invalid type for plugin {plugin}. Expected a precision or training type plugin."
                     )
-            if isinstance(plug, str):
-                # Reset the distributed type as the user has overridden training type
-                # via the plugins argument
-                self._strategy_type = None
-                self.set_distributed_mode(plug)
-
-            elif isinstance(plug, Strategy):
-                if strategy is None:
-                    strategy = plug
 
-                else:
+        # if user pass in a strategy class which has accelerator, precision, checkpoint or cluster env set up
+        if self._strategy_flag and isinstance(self._strategy_flag, Strategy):
+            if self._strategy_flag._accelerator:
```

Comment:
Base on the current lazy init way, we need check _property to avoid get default value. 
eg: https://github.com/PyTorchLightning/pytorch-lightning/blob/a64438c89717e4115d0f7cd4e8f93a812575a704/pytorch_lightning/strategies/strategy.py#L72-L74
Not for accelerator, but better to be consistent

Reply:
The proper lazy initialization will change all this, will be much easier to read